analyzers/src/SonarAnalyzer.CSharp/Rules/ConditionalSimplification.cs

```diff
@@ -89,16 +110,38 @@ protected override void Initialize(SonarAnalysisContext context)
             context.RegisterSyntaxNodeActionInNonGenerated(CheckConditionalExpression, SyntaxKind.ConditionalExpression);
             context.RegisterSyntaxNodeActionInNonGenerated(CheckIfStatement, SyntaxKind.IfStatement);
             context.RegisterSyntaxNodeActionInNonGenerated(CheckCoalesceExpression, SyntaxKind.CoalesceExpression);
+            context.RegisterSyntaxNodeActionInNonGenerated(CheckNotPattern, SyntaxKindEx.NotPattern);
+        }
+
+        private static void CheckNotPattern(SyntaxNodeAnalysisContext context)
+        {
+            var wrapper = (UnaryPatternSyntaxWrapper)context.Node;
+            if (wrapper.Pattern.IsNot()
+                && GetNegationRoot(context.Node) is var negationRoot
+                && negationRoot == context.Node)
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(RuleMultipleNegation, negationRoot.GetLocation()));
+            }
+
+            static SyntaxNode GetNegationRoot(SyntaxNode node)
+            {
+                while (node.Parent.Kind() == SyntaxKindEx.NotPattern)
+                {
+                    node = node.Parent;
+                }
+                return node;
+            }
         }
 
         private static void CheckCoalesceExpression(SyntaxNodeAnalysisContext context)
         {
-            if (context.Node.GetFirstNonParenthesizedParent() is AssignmentExpressionSyntax assignment && IsCoalesceAssignmentSupported(context))
+            if (context.Node.GetFirstNonParenthesizedParent() is AssignmentExpressionSyntax assignment
+                && context.Compilation.IsCoalesceAssignmentSupported())
             {
                 var left = ((BinaryExpressionSyntax)context.Node).Left.RemoveParentheses();
                 if (CSharpEquivalenceChecker.AreEquivalent(assignment.Left.RemoveParentheses(), left))
                 {
-                    context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, assignment.GetLocation(), BuildCodeFixProperties(context), "??="));
+                    context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, assignment.GetLocation(), BuildCodeFixProperties(context), "??="));
```

Comment:
Maybe it would be nice to create constants for all the operator strings, not to repeat them so often.